### PR TITLE
telemetry: use native stats by default

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Follow up to https://github.com/istio/istio/pull/41441 but for non-telemetry API installation (default), until a plan to default to telemetry API emerges. This should give sufficient burn time in CI, and avoid support for rarely used Wasm version.

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
LGTM just wanted to verify - are all the overrides possible compatible with native vs WASM? Or are there some obscure parts of the config that are not implemented/compatible?

**Comment by kyessenov** [comment]
@howardjohn What exactly do you mean by overrides? Some CEL expressions that refer to Wasm attributes are not compatible, because it's not Wasm. Those are not documented, so unlikely to actually be in-use.

**Comment by howardjohn** [comment]
I mean the `.Values.telemetry.v2.prometheus.configOverride.outboundSidecar`. I think we had established all fields exposed via Telemetry are compatible, but the configOverride can access arbitrary fields in the internal proto.

**Comment by kyessenov** [comment]
The schema is the same for wasm and stats filter, but I marked some fields removed. Seems like istio fails on the removed xDS fields, so we need to set the flag.

**Comment by kyessenov** [comment]
@zirain argh, another metric for istio version needs to be added. not sure why this test fails now only.

**Comment by zirain** [comment]
> @zirain argh, another metric for istio version needs to be added. not sure why this test fails now only.

IC

```console
istioctl x es sleep-69cfb4968f-h72rz | grep .istio_build
wasmcustom.component.proxy.tag.1.15.3;.istio_build: 1
```

**Review by zirain — APPROVED** [review]
lgtm

**Review by hzxuzhonghu — COMMENTED** [review]
@kyessenov Why do we still need the envoyfilter? Couldn't move it to native implement?

**Comment by kyessenov** [comment]
@hzxuzhonghu We may not need this if https://github.com/istio/istio/issues/37645 gets done, but noone is assigned to that issue, so until then, this will give us some burn time with EnvoyFilters.

**Review by hzxuzhonghu — APPROVED** [review]

File: pkg/envoy/proxy.go [hunk 1]
```diff
@@ -130,6 +130,7 @@ func (e *envoy) args(fname string, bootstrapConfig string) []string {
 		// than the flush interval.
 		"--file-flush-interval-msec", "1000",
 		"--disable-hot-restart", // We don't use it, so disable it to simplify Envoy's logic
+		"--allow-unknown-static-fields",
```
Comment (howardjohn) [inline pkg/envoy/proxy.go:133]: Why was this changed? Are there unknown fields now that we are accepting?

Comment (kyessenov) [reply pkg/envoy/proxy.go:133]: Yeah, because Wasm used a JSON parser that ignored fields with typos so I'm fairly sure some metric overrides had ill-defined fields.

Comment (kyessenov) [reply pkg/envoy/proxy.go:133]: Why "static" field flags affects ECDS is still not clear to me, likely a bug somewhere.

Comment (howardjohn) [reply pkg/envoy/proxy.go:133]: I am a bit worried we are going to miss real bugs now for this small telemetry case. But at the same time it would not be great if users with broken overrides pod's don't startup when they upgrade.

On the other hand, they would _probably_ immediately find the issue and rollback/fix I would imagine if we did hard fail...

Comment (kyessenov) [reply pkg/envoy/proxy.go:133]: We still get a warning for unknown fields.

I think we have to support unknown fields for automatic upgrades, there's just no other way to introduce new fields without putting conditions everywhere.

Comment (howardjohn) [reply pkg/envoy/proxy.go:133]: Well this is only for static config, right? So over XDS we are fine; static is tightly coupled with the Envoy version since the bootsrtap is generated by pilot-agent which is in the same container... except now we have bootstrap DS so maybe this makes sense now.

I am a bit confused how this dynamic XDS is impacted by `allow-unknown-static-fields` which I thought was only for startup bootstrap config

Comment (kyessenov) [reply pkg/envoy/proxy.go:133]: > I am a bit confused how this dynamic XDS is impacted by `allow-unknown-static-fields` which I thought was only for startup bootstrap config

Me, too, it's probably a bug but it's already shipped so we have to live with it.

Comment (howardjohn) [reply pkg/envoy/proxy.go:133]: Fair enough. If its a bug might be worth filing an envoy issue? but this lgtm for now - even with it fixed, for Bootstrap DS we may want it.
